### PR TITLE
Enable Online Tests with GitHub Actions

### PR DESCRIPTION
**Pull request by omus**
Note I did my experimentation in a [separate PR](https://github.com/JuliaCloud/AWSClusterManagers.jl/pull/5) which became very messy due to triggering bors

### CODE REVIEW

**Comment by omus** [comment]
bors try

**Comment by omus** [comment]
Cancelled the workflow as I didn't enable all of the online tests.

**Comment by omus** [comment]
bors try

**Comment by omus** [comment]
Just going to mention that GHA could be better when viewing active workflows. I've been watching the logs and it appears to be generating the system image but I've since noticed AWS Batch jobs being submitted so it seems that the log output has just stalled

**Comment by omus** [comment]
There's an issue when running on AWS Batch when GHA has used `CREATE_SYSIMG=true` when building the Docker image. Disabling this seems to work fine so I'll just open an issue to avoid holding this PR up: https://github.com/JuliaCloud/AWSClusterManagers.jl/issues/9

**Comment by omus** [comment]
bors try

**Comment by omus** [comment]
The "Fix CloudWatch logs ingestion check" fixes https://github.com/JuliaCloud/AWSClusterManagers.jl/issues/10

**Comment by omus** [comment]
bors try

File: src/container.jl [hunk 1]
```diff
@@ -124,7 +124,8 @@ end
 
 # https://github.com/aws/amazon-ecs-agent/issues/1119
 # Note: For AWS Batch array jobs the "ecs/<job_id>" does not include the array index
-const CGROUP_REGEX = r"\/(?:docker|ecs\/[0-9a-f]{32})\/(?<container_id>[0-9a-f]{64})\b"
+# Note: GitHub Actions uses "actions_job"
+const CGROUP_REGEX = r"\/(?:docker|ecs\/[0-9a-f]{32}|actions_job)\/(?<container_id>[0-9a-f]{64})\b"
```
Comment (mattBrzezinski) [inline src/container.jl:128]: Could we leave a comment explaining the regex, and an example of what's to match?

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -53,6 +53,72 @@ jobs:
...
+        run: aws cloudformation validate-template --template-body file://test/batch.yml
+
+      # Note: Using Bors will restrict us to only having one workflow running at a given time.
+      # Because of this we can safely use a single CFN stack and just update the stack to the
+      # state of the current branch.
+      - name: Deploy to AWS CloudFormation
+        uses: aws-actions/aws-cloudformation-github-deploy@v1
+        with:
+          name: ${{ env.STACK_NAME }}
+          template: test/batch.yml
```
Comment (mattBrzezinski) [inline .github/workflows/CI.yml:90]: Can we not have this stack just living constantly?

Comment (omus) [reply .github/workflows/CI.yml:90]: That is an alternate way we could handle it but it does make it more difficult to make changes to the stack over time. For example when adding AWS Batch node support we needed to switch to an on-demand cluster vs. spot.

This way allows for contributors to this repo to make updates to the stack without having direct access to the AWS account.

File: src/container.jl (hunk 1 above)
Comment (omus) [reply src/container.jl:128]: I'll add an explanation here

File: .github/workflows/CI.yml (hunk 2 above)
Comment (mattBrzezinski) [reply .github/workflows/CI.yml:90]: Makes sense!

**Comment by omus** [comment]
Just going to mention that the `AWSClusterManagers-jl` user was created manually with the following permissions:
```json
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Action": "cloudformation:ValidateTemplate",
            "Resource": "*"
        },
        {
            "Effect": "Allow",
            "Action": "iam:PassRole",
            "Resource": "arn:aws:iam::134847318362:role/CloudFormationAdmin"
        },
        {
            "Effect": "Allow",
            "Action": [
                "sts:AssumeRole",
                "sts:TagSession"
            ],
            "Resource": "arn:aws:iam::134847318362:role/AWSClusterManagers-jl*",
        },
        {
            "Effect": "Allow",
            "Action": [
                "cloudformation:*"
            ],
            "Resource": "arn:aws:cloudformation:us-east-1:134847318362:stack/AWSClusterManagers-jl*/*"
        }
    ]
}
```

**Pull request merged**